core/src/solve.jl

```diff
@@ -521,15 +523,26 @@ function formulate_flow!(
         end
 
         inflow_id = inflow_edge.edge[1]
+        outflow_id = outflow_edge.edge[2]
+
         factor = low_storage_factor(u.storage, inflow_id, 10.0)
         q = flow_rate * factor
-        q = clamp(q, min_flow_rate, max_flow_rate)
 
-        set_flow!(graph, inflow_edge, q, du)
+        _, src_level = get_level(p, inflow_id, t, du)
```

Comment:
Good call, I refactored `get_level` to return only a number, which I set to -Inf for Terminal to avoid reduction factors based on downstream level checks.